tests/test_resolver.py

```diff
@@ -0,0 +1,129 @@
+"""Verify behavior of packages resolver"""
+
+from typing import Dict, List, Optional
+
+from hypothesis import HealthCheck, assume, given, settings
+from hypothesis import strategies as st
+
+from fawltydeps.packages import DependenciesMapping, Package, resolve_dependencies
+
+
+def dict_subset_strategy(input_dict):
+    keys_strategy = st.lists(st.sampled_from(list(input_dict.keys())), unique=True)
+
+    def build_dict(keys):
+        return {key: input_dict[key] for key in keys}
+
+    return keys_strategy.map(build_dict)
+
+
+# The deps in each category should be disjoint
+
+# non locally installed deps
+non_locally_installed_deps = ["pandas", "numpy", "other"]
+non_locally_installed_strategy = st.lists(
+    st.sampled_from(non_locally_installed_deps), unique=True
+)
+
+# locally installed deps
+locally_installed_deps = {
+    "setuptools": [
+        "_distutils_hack",
+        "pkg_resources",
+        "setuptools",
+    ],
+    "pip": ["pip"],
+    "isort": ["isort"],
+}
+locally_installed_strategy = dict_subset_strategy(locally_installed_deps)
+
+# user-defined deps
+user_defined_deps = {"apache-airflow": ["airflow"]}
+user_defined_strategy = dict_subset_strategy(user_defined_deps)
+
+# user-defined mapping
+user_defined_mapping = "\n".join(
+    [f'{dep} = ["{",".join(imports)}"]' for dep, imports in user_defined_deps.items()]
+)
+
+user_mappings = st.one_of(st.none(), st.just(user_defined_mapping))
+
+
+def generate_expected_resolved_deps(
+    locally_installed_deps: Optional[Dict[str, List[str]]] = None,
+    non_locally_installed_deps: Optional[List[str]] = None,
+    user_defined_deps: Optional[Dict[str, List[str]]] = None,
+):
+    ret = {}
+    if locally_installed_deps:
+        ret.update(
+            {
+                dep: Package(
+                    dep,
+                    {DependenciesMapping.LOCAL_ENV: set(imports)},
+                )
+                for dep, imports in locally_installed_deps.items()
+            }
+        )
+    if user_defined_deps:
+        ret.update(
+            {
+                dep: Package(
+                    dep,
+                    {DependenciesMapping.USER_DEFINED: set(imports)},
+                )
+                for dep, imports in user_defined_deps.items()
+            }
+        )
+    if non_locally_installed_deps:
+        ret.update(
+            {
+                dep: Package(dep, {DependenciesMapping.IDENTITY: {dep}})
+                for dep in non_locally_installed_deps
+            }
+        )
+    return ret
+
+
+# Suppressing the warning on function scoped fixtures: even if the tmp writing
+# fixture only runs once for all the test cases, it will be writing the same
+# content
```

Comment:
:+1:
I would be more explicit relating to docummentation:
```suggestion
# Suppressing the warning on function scoped fixtures: even if the tmp writing
# fixture only runs once for all the test cases, it is not a problem,
# as the fixture-generated content does not have to be reset between test examples.
# The test function only reads the file content and filters needed input.
```

Reply:
Updated in [2e3b6db](https://github.com/tweag/FawltyDeps/pull/296/commits/2e3b6db8f89dd4294fb3387c06e48803a463a31d).